django/db/models/expressions.py

```diff
@@ -686,22 +686,25 @@ class Col(Expression):
 
     contains_column_references = True
 
-    def __init__(self, alias, target, output_field=None):
+    def __init__(self, alias, target, output_field=None, index_col=False):
         if output_field is None:
             output_field = target
         super().__init__(output_field=output_field)
-        self.alias, self.target = alias, target
+        self.alias, self.target, self.index_col = alias, target, index_col
 
     def __repr__(self):
         return "{}({}, {})".format(
             self.__class__.__name__, self.alias, self.target)
 
     def as_sql(self, compiler, connection):
+        if self.index_col:
+            return connection.ops.quote_name(self.alias), []
```

Comment:
hmm alias generally refers to the table name - not the column name. You're changing the meaning of alias depending on the index_col.

Maybe just pass `None, 'name', index_col=True` ?